### PR TITLE
Remove redundant `unalias_nested` calls for creating Filter's

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/11339

## Rationale for this change

In https://github.com/apache/datafusion/pull/11307 we changed Filter to remove any aliases within the expressions.

Thus it is no longer necessary to unalias the expressions prior to calling `Filter::try_new`

## What changes are included in this PR?

1. Remove the `unalias_nested` calls when calling `Filter::try_new()`

## Are these changes tested?
By existing CI

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/expr/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -645,39 +643,6 @@ impl LogicalPlan {
                 Ok(LogicalPlan::Values(Values { schema, values }))
             }
             LogicalPlan::Filter(Filter { predicate, input }) => {
-                // todo: should this logic be moved to Filter::try_new?
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:648]: this PR completes this TODO, per @jonahgao 's suggestion

File: datafusion/optimizer/src/push_down_filter.rs [hunk 2]
```diff
@@ -761,11 +761,11 @@ impl OptimizerRule for PushDownFilter {
 
                 // Push down non-unnest filter predicate
                 // Unnest
-                //   Unenst Input (Projection)
+                //   Unnest Input (Projection)
```
Comment (alamb) [inline datafusion/optimizer/src/push_down_filter.rs:764]: drive by typo fixup

File: datafusion/optimizer/src/common_subexpr_eliminate.rs [hunk 3]
```diff
@@ -344,12 +344,10 @@ impl CommonSubexprEliminate {
...
             .transform_data(|(mut new_expr, new_input)| {
                 assert_eq!(new_expr.len(), 1); // passed in vec![predicate]
-                let new_predicate = new_expr
-                    .pop()
-                    .unwrap()
-                    .unalias_nested()
-                    .update_data(|new_predicate| (new_predicate, new_input));
-                Ok(new_predicate)
+                let new_predicate = new_expr.pop().unwrap();
+                // the closure didn't transform the expr, but will be
```
Comment (jonahgao) [inline datafusion/optimizer/src/common_subexpr_eliminate.rs:348]: Is it possible to remove `transform_data`?
```rust
self.try_unary_plan(expr, input, config)?
            .map_data(|(mut new_expr, new_input)| {
                assert_eq!(new_expr.len(), 1); // passed in vec![predicate]
                let new_predicate = new_expr.pop().unwrap();
                Filter::try_new(new_predicate, Arc::new(new_input))
                    .map(LogicalPlan::Filter)
            })
```

**Review by jonahgao — APPROVED** [review]
Looks good to me! Thanks @alamb

File: datafusion/optimizer/src/common_subexpr_eliminate.rs (hunk 3 above)
Comment (alamb) [reply datafusion/optimizer/src/common_subexpr_eliminate.rs]: That is an excellent idea. It was -- done in 49d206e65

**Comment by alamb** [comment]
Thanks again @jonahgao

**Pull request merged**